### DISCUSSION TITLE
Bridge useAsyncData composable

### DISCUSSION THREAD

**Discussion by Rigo-m** [nuxt/nuxt#18305] [Questions]
Hi guys, so happy with nuxt 3 being released!

I tracked the `bridge-edge` repo closely the last few weeks, because I was very curious on `useAsyncData` being implemented (especially with the `defer: true` capability) in Nuxt2.
However, I see that the composable is still mocked out in current Bridge release.

Since we have hard limitations on Vue 2 (`async setup` not possible, no `<Suspense>` component etc.), are you willing to share when (but most possibly how) a deferred useAsyncData will be implemented in Bridge?

Thanks,

Matteo

**Comment by pi0** [comment]
Hi @Rigo-m. Indeed we were internally discussing how to support `useAsyncData` for Bridge with @danielroe and main blocker is Suspense support for vue2. On the other hand, we don't want to advise using `useAsyncData` without await in docs so a similar implementation (non-blocking) will be added to bridge (with a different name) but we can also use it as a partial shim for bridge.

**Reply by Rigo-m** [reply]
So basically this is saying that there will **never** be a blocking implementation in nuxt2? Or is there still some hope?

**Comment by Rigo-m** [comment]
It would be cool to have it at least as a composable that we can use in a `Page` component, much like asyncData but inside setup in the page

**Comment by Rigo-m** [comment]
@pi0 @danielroe I am currently on a nuxt2 + nuxtjs/composition API big project, and handling SSR has been really tough, especially because it's hard to handle skeletons/loaders without a `<Suspense>` component. I was hoping bridge would be helpful to solve this issue. All I need is a way to defer data fetching a-la-asyncData at least in page components, so I can retain my `setup` setup without rewriting everything in `asyncData` or skeleton-ize every component.
Please help me understand what are my options, I would be really glad to know.

As always, thanks for the hard job you're putting into the nuxt community.
Have a nice day
